mitxgraders/formulagrader/matrixgrader.py

```diff
@@ -88,15 +92,27 @@ def check_response(self, answer, student_input, **kwargs):
             with MathArray.enable_negative_powers(self.config['negative_powers']):
                 result = super(MatrixGrader, self).check_response(answer, student_input, **kwargs)
         except ShapeError as err:
-            if self.config['shape_errors']:
+            if self.config['suppress_matrix_messages']:
+                return {'ok': False, 'msg': '', 'grade_decimal': 0}
+            elif self.config['shape_errors']:
                 raise
             else:
                 return {'ok': False, 'msg': err.message, 'grade_decimal': 0}
         except InputTypeError as err:
-            if self.config['answer_shape_mismatch']['is_raised']:
+            if self.config['suppress_matrix_messages']:
+                return {'ok': False, 'msg': '', 'grade_decimal': 0}
+            elif self.config['answer_shape_mismatch']['is_raised']:
                 raise
             else:
                 return {'ok': False, 'grade_decimal': 0, 'msg': err.message}
+        except (DomainError, MathArrayError) as err:
```

Comment:
This will also suppress some errors that are not matrix-related, like `sin(1, 3)` or `kronecker(5)` (though I'm not particularly worried about inputs like that, and the result is just marking them wrong, which is reasonable). If you're OK with that, I'm happy to merge.

Reply:
Hmm. That shouldn't be all that hard to fix, honestly. Let me see if I can do so...

Reply:
Ok, I think I've fixed this. It required cleaning up our exception structure a little though. Take another look?

Reply:
👍 Nice idea, looks good.